ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -146,15 +147,18 @@ private void processHead(Optional<UInt64> nodeSlot) {
                       forkChoiceStrategy.onAttestation(transaction, indexedAttestation));
           return transaction
               .commit()
-              .thenRun(() -> updateForkChoiceForImportedBlock(block, forkChoiceStrategy, result))
+              .thenRun(
+                  () ->
+                      updateForkChoiceForImportedBlock(
+                          block, forkChoiceStrategy, (SuccessfulBlockImportResult) result))
```

Comment:
Should avoid the cast to `SuccessfulBlockImportResult`.  Might need to make `setBlockOnCanonicalChain` a `markAsCanonical` method which throws `UnsupportedOperationException` for failed results.